ci/ray_ci/docker_container.py

```diff
@@ -48,13 +50,52 @@ def run(self) -> None:
         if self.python_version == "py37":
             constraints_file = "requirements_compiled_py37.txt"
 
-        version_tag = os.environ["BUILDKITE_COMMIT"][:6]
-        project = f"rayproject/{self.image_type}"
-        ray_image = f"{project}:{version_tag}-{self.python_version}-{self.platform}"
-
-        self.run_script(
-            [
-                "./ci/build/build-ray-docker.sh "
-                f"{wheel_name} {base_image} {constraints_file} {ray_image}"
+        ray_images = self._get_image_names()
+        ray_image = ray_images[0]
+        cmds = [
+            "./ci/build/build-ray-docker.sh "
+            f"{wheel_name} {base_image} {constraints_file} {ray_image}"
+        ]
+        if self._should_upload():
+            cmds += [
+                "pip install -q aws_requests_auth boto3",
+                "python .buildkite/copy_files.py --destination docker_login",
             ]
-        )
+            for alias in self._get_image_names():
+                cmds += [
+                    f"docker tag {ray_image} {alias}",
+                    f"docker push {alias}",
+                ]
+        self.run_script(cmds)
+
+    def _should_upload(self) -> bool:
+        return os.environ["BUILDKITE_PIPELINE_ID"] == POSTMERGE_PIPELINE
```

Comment:
could you use `os.environ.get()`, so that it still runs outside buildkite?

same with the ones inside `_get_images_names()`

Reply:
totally